expression/builtin_arithmetic_vec.go

```diff
@@ -400,12 +393,14 @@ func (b *builtinArithmeticMinusIntSig) minusFUS(result *chunk.Column, lhi64s, rh
 		}
 		lh, rh := lhi64s[i], rhi64s[i]
 
-		if lh < 0 {
-			return types.ErrOverflow.GenWithStackByArgs("BIGINT UNSIGNED", fmt.Sprintf("(%s - %s)", b.args[0].String(), b.args[1].String()))
-		}
-
-		if (lh > 0 && -rh > math.MaxInt64-lh) || (lh < 0 && -rh < math.MinInt64-lh) {
-			return types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("(%s + %s)", b.args[0].String(), b.args[1].String()))
+		// lh >= 0 && rh >= 0 : never overflow
```

Comment:
Sure we can, but I am not sure whether introducing extra function calls will worsen performance. I chose not to abstract these code into function, because in some other `builtinArithmetics` that met similar situations, `builtinArithmeticModIntSig` for instance, they did not abstract the check logic into function as well.

Reply:
It'll be hard to maintain the code in the future if we do not extract the code.

Reply:
I see, so do you think we should also extract overflow detection logic in other builtin functions as well, `builtinArithmeticModIntSig` for instance?

Reply:
We can do it in another PR is ok.
We'd better make a PR as small as possible to make it easy to be reviewed.